sanic/server.py

```diff
@@ -167,12 +167,18 @@ def on_message_complete(self):
     # -------------------------------------------- #
 
     def write_response(self, response):
+        keep_alive = (
```

Comment:
I don't want to catch that attribute error in the same try/except. I'm also thinking about bumping the parsing out of the protocol into another separate function where the HTTPRequest basically builds itself.